File: src/main/java/hiiragi283/material/api/capability/HiiragiCapability.java
package hiiragi283.material.api.capability;

import hiiragi283.material.api.capability.material.IMaterialHandler;
import hiiragi283.material.api.capability.material.MaterialHandler;
import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.CapabilityInject;
import net.minecraftforge.common.capabilities.CapabilityManager;

public abstract class HiiragiCapability {

    @CapabilityInject(IMaterialHandler.class)
    public static Capability<IMaterialHandler> MATERIAL_HANDLER_CAPABILITY;

    public static void register() {
        CapabilityManager.INSTANCE.register(IMaterialHandler.class, new HiiragiStorage<>(), () -> new MaterialHandler(1000, 1));
    }

}

File: src/main/kotlin/hiiragi283/api/capability/HiiragiCapability.kt
package hiiragi283.api.capability

import hiiragi283.api.capability.material.IMaterialHandler
import hiiragi283.api.capability.material.MaterialHandler
import net.minecraftforge.common.capabilities.Capability
import net.minecraftforge.common.capabilities.CapabilityInject
import net.minecraftforge.common.capabilities.CapabilityManager

object HiiragiCapability {

    @CapabilityInject(IMaterialHandler::class)
    lateinit var MATERIAL: Capability<IMaterialHandler>

    fun register() {

        CapabilityManager.INSTANCE.register(
            IMaterialHandler::class.java,
            HiiragiStorage<IMaterialHandler>()
        ) { MaterialHandler(capacity = 1000) }

    }

}